### PR TITLE
ts_calibrate: properly close framebuffer and touchscreen in validation mode

### LINKED ISSUE #142: ts_calibrate not closing fb/ts properly in validation mode
Hi,
I want to discuss the expected behavior of ts_calibrate when used in --validate mode.

The application now stays visually open after completion showing "passed/failed" without calling close_framebuffer() and ts_close() which might impact the system after use of ts_calibrate [ -> e.g. chvt hangs when ts_calibrate doesn't close the framebuffer properly].
I would like ts_calibrate --validate to close everything correctly.

I committed the bare necessary changes in #https://github.com/pfink-christ/tslib/commit/09808c632a5fd44e5b08e481e9cc9359ed051981 but this seems to be a quick shot solution only. This is because the final screen showing "passed/failed" cannot be seen anymore.

Maybe we could add a timeout (configurable with an option or hard coded to something like 5s). Or is the expected behavior to keep this screen forever (like it is now)?

What are your opinions?

I can make a pull request using my already committed solution or implement the behavior we agree on here.

Thanks,
Peter

### PR DESCRIPTION
**Pull request by pfink-christ**
This closes #142 as discussed.

### CODE REVIEW

**Comment by merge** [comment]
thanks a lot. give me a week to review this.

Am 28. Dezember 2018 15:21:49 MEZ schrieb Peter Fink <notifications@github.com>:
>This closes #142 as discussed.
>You can view, comment on, or merge this pull request online at:
>
>  https://github.com/kergoth/tslib/pull/143
>
>-- Commit Summary --
>
>  * tests: ts_calibrate.c: close framebuffer and ts after validation
>  * tests: ts_calibrate: add new timeout option for validate mode
>* tests: ts_calibrate: change default timeout to 5s as dicussed in #142
>
>-- File Changes --
>
>    M tests/ts_calibrate.c (35)
>
>-- Patch Links --
>
>https://github.com/kergoth/tslib/pull/143.patch
>https://github.com/kergoth/tslib/pull/143.diff

-- 
Martin Kepplinger
http://martinkepplinger.com
sent from mobile

File: tests/ts_calibrate.c [hunk 1]
```diff
@@ -182,6 +182,12 @@ static int ts_validate(struct tsdev *ts, int boundary, unsigned int loops)
 	fillrect(0, 0, xres - 1, yres - 1, 0);
 	put_string_center(xres / 2, yres / 4, textbuf, 1);
 
+	if(timeout == 0){
```
Comment (merge) [inline tests/ts_calibrate.c]: please fix the coding style

Comment (pfink-christ) [reply tests/ts_calibrate.c]: Please recheck.

Comment (merge) [reply tests/ts_calibrate.c]: still, a space between if and the open bracket seems to be missing...

also, could you squash your 3 changes to be only one commit? It all is one change, adding the -s option that closes the framebuffer.

other than that, I like it. Thanks a lot!

Comment (pfink-christ) [reply tests/ts_calibrate.c]: sorry I missed that completely ;)

**Comment by pfink-christ** [comment]
squashed the commits into one

**Pull request closed without merging**